src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -307,6 +332,96 @@ internal Silo(string name, SiloType siloType, ClusterConfiguration config, ILoca
             logger.Info(ErrorCode.SiloInitializingFinished, "-------------- Started silo {0}, ConsistentHashCode {1:X} --------------", SiloAddress.ToLongString(), SiloAddress.GetConsistentHashCode());
         }
 
+        private IServiceCollection RegisterSystemTypes()
+        {
+            //
+            // Register the system classes and grains in this method.
+            //
+
+            IServiceCollection serviceCollection = new ServiceCollection();
+
+            serviceCollection.AddTransient<ManagementGrain>();
+            serviceCollection.AddTransient<GrainBasedMembershipTable>();
+            serviceCollection.AddTransient<GrainBasedReminderTable>();
+            serviceCollection.AddTransient<PubSubRendezvousGrain>();
+            serviceCollection.AddTransient<MemoryStorageGrain>();
+
+            return serviceCollection;
+        }
+
+        private void HandleStartup(IServiceCollection serviceCollection)
+        {
+            if (String.IsNullOrWhiteSpace(nodeConfig.StartupTypeName))
+            {
+                return;
+            }
+
+            var startupType = System.Type.GetType(nodeConfig.StartupTypeName);
+
+            if (startupType == null)
+            {
+                throw new InvalidOperationException(string.Format("Can not locate the type specified in the configuration file: '{0}'.", nodeConfig.StartupTypeName));
+            }
+
+            var servicesMethod = FindConfigureServicesDelegate(startupType);
```

Comment:
I'm using Asp.Net vNext as a reference, not the old owin based one.

It supports 3 type of startup class behavior, but as I sad before, Under Asp.Net they're handling the binaries differently and they preloading all assemblies into memory, so Attribute based discovery can work. Since Orleans - currently - works otherwise we can't do that.

This feature is a starter which can be evolved as time goes on.